stdlib/messaging/activemq-artemis/src/main/java/org/ballerinalang/messaging/artemis/ArtemisServiceCompilerPlugin.java

```diff
@@ -65,73 +61,39 @@ public void process(ServiceNode serviceNode, List<AnnotationAttachmentNode> anno
                 count++;
             }
         }
-
         if (count == 0) {
             dlog.logDiagnostic(Diagnostic.Kind.ERROR, serviceNode.getPosition(),
-                               "There has to be an artemis:ServiceConfig annotation declared for service");
+                    "There has to be an artemis:ServiceConfig annotation declared for service");
         }
 
         @SuppressWarnings(ArtemisConstants.UNCHECKED)
         List<BLangFunction> resources = (List<BLangFunction>) serviceNode.getResources();
-        if (resources.size() > 1) {
+        // More than two resource functions are found
+        if (resources.size() > 2) {
             dlog.logDiagnostic(Diagnostic.Kind.ERROR, serviceNode.getPosition(),
-                               "Only one resource is allowed in the service");
-        }
-        validate(resources.get(0));
-    }
-
-    private void validate(BLangFunction resource) {
-        if (!isResourceReturnsErrorOrNil(resource)) {
-            dlog.logDiagnostic(Diagnostic.Kind.ERROR, resource.pos, "Invalid return type: expected error?");
-        }
-        List<BLangSimpleVariable> paramDetails = resource.getParameters();
-        if (paramDetails == null || paramDetails.isEmpty() || paramDetails.size() > 2) {
-            dlog.logDiagnostic(Diagnostic.Kind.ERROR, resource.pos,
-                               INVALID_RESOURCE_SIGNATURE_FOR + resource.getName().getValue() +
-                                       " resource: Unexpected parameter count(expected parameter count 1 or 2)");
-            return;
-        }
-        if (!ArtemisConstants.MESSAGE_OBJ_FULL_NAME.equals(paramDetails.get(0).type.toString())) {
-            dlog.logDiagnostic(Diagnostic.Kind.ERROR, resource.pos, INVALID_RESOURCE_SIGNATURE_FOR
-                    + resource.getName().getValue() + " resource: The first parameter should be an artemis:Message");
-        }
-        if (paramDetails.size() == 2) {
-            validateSecondParam(resource, paramDetails);
-        }
-    }
-
-    private void validateSecondParam(BLangFunction resource, List<BLangSimpleVariable> paramDetails) {
-        BType secondParamType = paramDetails.get(1).type;
-        int secondParamTypeTag = secondParamType.tag;
-        if (secondParamTypeTag != TypeTags.STRING && secondParamTypeTag != TypeTags.JSON &&
-                secondParamTypeTag != TypeTags.XML && secondParamTypeTag != TypeTags.RECORD &&
-                secondParamTypeTag != TypeTags.MAP && checkArrayType(secondParamType)) {
-            dlog.logDiagnostic(Diagnostic.Kind.ERROR, resource.pos, INVALID_RESOURCE_SIGNATURE_FOR
-                    + resource.getName().getValue() + " resource in service " +
-                    ": The second parameter should be a string, json, xml, byte[], map or a record type");
+                    "Only maximum of two resources are allowed in the service");
```

Comment:
Shall we wording this as only onMessage and onError are allowed etc?

Reply:
Resolved